packages/react/menu/src/Menu.tsx

```diff
@@ -31,17 +31,48 @@ const ALL_KEYS = [...FIRST_KEYS, ...LAST_KEYS];
 const MENU_NAME = 'Menu';
 
 type MenuContextValue = {
-  menuRef: React.RefObject<HTMLDivElement>;
-  onItemsReachableChange(open: boolean): void;
+  open: boolean;
   onOpenChange(open: boolean): void;
 };
+
 const [MenuProvider, useMenuContext] = createContext<MenuContextValue>(MENU_NAME);
 
-type MenuOwnProps = Polymorphic.Merge<
-  Polymorphic.OwnProps<typeof MenuImpl>,
-  {
-    open?: boolean;
+type MenuOwnProps = {
+  open?: boolean;
+  onOpenChange?(open: boolean): void;
+};
 
+const Menu: React.FC<MenuOwnProps> = (props) => {
+  const { open = false, children, onOpenChange } = props;
+  const handleOpenChange = useCallbackRef(onOpenChange);
```

Comment:
Same on this one, I think we usually use `React.useCallback` before passing into providers no?

Reply:
Yeah we do that when passing an internal function, but this is a prop. If the consumer passes an inline function as the `onOpenChange` prop then the context consumers would all constantly re-render, even if I do `React.useCallback(onOpenChange, [onOpenChange])`.